src/CrvInfoHelper.cc

```diff
@@ -37,12 +37,34 @@ namespace mu2e
     {
       const CrvCoincidenceCluster &cluster = crvCoincidences->at(i);
 
+      // Get the PEs per layer from the reco pulses
+      float PEsPerLayer_[4] = {0.}; // Four layers, each one initiliased to zero PEs in a static array
+      const std::vector<art::Ptr<CrvRecoPulse> > coincRecoPulses_ = cluster.GetCrvRecoPulses(); // Get the reco pulses from the coincidence 
+      for(size_t j=0; j<coincRecoPulses_.size(); j++) // Loop through the pulses
+      {
+        // Skip duplicate pulses (those with multiple peaks)
+        if(coincRecoPulses_.at(j)->GetRecoPulseFlags().test(CrvRecoPulseFlagEnums::duplicateNoFitPulse)) continue;
+        // Get layer number from the bar index associated with this reco pulse
+        const CRSScintillatorBarIndex &crvBarIndex = coincRecoPulses_.at(j)->GetScintillatorBarIndex(); 
+        const CRSScintillatorBar &crvCounter = CRS->getBar(crvBarIndex);
+        const CRSScintillatorBarId &crvCounterId = crvCounter.id();
+        int layerNumber = crvCounterId.getLayerNumber();
+        // Sum PEs for this coincidence, indexed by layer number
+        PEsPerLayer_[layerNumber] += coincRecoPulses_.at(j)->GetPEsNoFit(); // The coincidences were found using the NoFit option, so use that here as well  
+      }
+      // Sanity check for PEsPerLayer
+      float PEsPerLayerSum = 0.; 
+      for (size_t j=0; j<4; j++) PEsPerLayerSum += PEsPerLayer_[j];
+      float deltaPEs = abs(PEsPerLayerSum - cluster.GetPEs());
+      if (deltaPEs > 1e-3) std::cout<<"CrvInfoHelper warning: the sum of PEs per layer differs by " << deltaPEs << " from total PEs in the coincidence!"<<std::endl;
```

Comment:
Is this warning bad enough that we should really throw an exception and stop execution?

Reply:
Maybe, although the `1e-3` limit is just something I made up, which is the main reason I made it a `cout` rather than a `cerr`. To be honest I don't understand why there is any difference between the two values at all. 

There is a tricky detail where you need to change the way you get your PEs from the reco pulses depending on the way they were reconstructed in the coincidence. There are a few different ways PEs are formed, see [Offline/CrvCoincidenceFinder_module.cc#L314-L317](https://github.com/Mu2e/Offline/blob/main/CRVReco/src/CrvCoincidenceFinder_module.cc#L314-L317), and I was originally using `GetPEs()` rather than `GetPEsNoFit()`, but was seeing a large (like 5-10 PEs) discrepancy between the total PEs from the coincidence and the sum of the PEs from the reco pulses, which is why I originally made that printout. 

Maybe we need to discuss this with Ralf?

Reply:
Yes, I think it would be good to discuss with Ralf. It would be good if TrkAna didn't need to know how the PEs were formed and could just use a ```GetPEs()``` function and be given the correct ones. Then we don't need to update anything if the method changes in the CRV reconstruction code